### DISCUSSION TITLE
Why "value of `paths.{}` should be an array with one element because the src path does not contains * (wildcard)" ?

### DISCUSSION THREAD

**Discussion by obzva** [swc-project/swc#8396] [Q&A] [answered]
# TLDR
What does "value of `paths.{}` should be an array with one element because the src path does not contains * (wildcard)" mean and why should it be?
***
Hi I'm working on old Next.js project and I recently upgraded it to v13.
I ran some jest test on my project and met some of this fail log like below.
```
  ● Test suite failed to run

    failed to process assertion failed: `(left == right)`
      left: `2`,
     right: `1`: value of `paths.@components` should be an array with one element because the src path does not contains * (wildcard)

      at Object.transformSync (node_modules/next/src/build/swc/index.ts:1000:25)
      at transformSync (node_modules/next/src/build/swc/index.ts:1181:19)
      at Object.process (node_modules/next/src/build/swc/jest-transformer.ts:117:25)
      at ScriptTransformer.transformSource (node_modules/@jest/transform/build/ScriptTransformer.js:542:31)
      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:671:40)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:723:19)
```
This failure is not from the test code itself because my code looks like below.
```
describe('Test', () => {
  it('this should pass', () => {
    const a = 1;
    expect(a).toBe(1);
  });
});
```
So I checked the tsconfig.json file and amended some.
```
from
      "@components/*": ["components/*"],
      "@components": ["components"],

to
      "@components/*": ["src/components/*", "components/*"],
      "@components": ["src/components", "components"],
```
And it worked well!

I wanted to know the detail of this failure and rules of path aliasing, so I digged typescript docs but actually I found that it is on SWC.
I couldn't understand what is the message and by typescript docs, even if the first path is invalid, the second one will be the fallback so I thought it is OK.

**Comment by kdy1** [comment, marked as answer]
I think it's a bug

**Reply by obzva** [reply]
Okay, you mean it's a bug of SWC right?

**Reply by galczo5** [reply]
@kdy1 do you have any link to pull request or an issue that solves this problem?

**Reply by kdy1** [reply]
No, I don't think they filed one

**Comment by tonyganchev** [comment]
Hitting this issue in a nx@20 environment with a build using _@nx/angular:webpack-browser_ executor.

```
⠙ Generating browser application bundles (phase: setup)...thread '<unnamed>' panicked at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/swc_ecma_loader-7.0.0/src/resolvers/tsc.rs:83:21:
assertion `left == right` failed: value of `paths.[scope]/[package]/[2ary-entry-point]` should be an array with one element because the src path does not contains * (wildcard)
  left: 3
 right: 1
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

 NX   failed to handle: assertion `left == right` failed: value of `paths.[scope]/[package]/[2ary-entry-point]` should be an array with one element because the src path does not contains * (wildcard)

  left: 3
 right: 1
Error: failed to handle: assertion `left == right` failed: value of `paths.[scope]/[package]/[2ary-entry-point]` should be an array with one element because the src path does not contains * (wildcard)
  left: 3
 right: 1
    at Compiler.transformSync (/home/[redacted]/src/[redacted]/node_modules/.pnpm/@swc+core@1.10.18/node_modules/@swc/core/index.js:249:29)
    at transformSync (/home/[redacted]/src/[redacted]/node_modules/.pnpm/@swc+core@1.10.18/node_modules/@swc/core/index.js:359:21)
    at transformSync (/home/[redacted]/src/[redacted]/node_modules/.pnpm/@swc-node+core@1.13.3_@swc+core@1.10.18_@swc+types@0.1.17/node_modules/@swc-node/core/index.ts:93:26)
    at compile (/home/[redacted]/src/[redacted]/node_modules/.pnpm/@swc-node+register@1.10.9_@swc+core@1.10.18_@swc+types@0.1.17_typescript@5.4.5/node_modules/@swc-node/register/register.ts:111:40)
    at /home/[redacted]/src/[redacted]/node_modules/.pnpm/@swc-node+register@1.10.9_@swc+core@1.10.18_@swc+types@0.1.17_typescript@5.4.5/node_modules/@swc-node/register/register.ts:122:38
    at Module._compile (/home/[redacted]/src/[redacted]/node_modules/.pnpm/pirates@4.0.6/node_modules/pirates/lib/index.js:113:29)
    at node:internal/modules/cjs/loader:1706:10
    at Object.newLoader [as .ts] (/home/[redacted]/src/[redacted]/node_modules/.pnpm/pirates@4.0.6/node_modules/pirates/lib/index.js:121:7)
    at Module.load (node:internal/modules/cjs/loader:1289:32)
    at Function._load (node:internal/modules/cjs/loader:1108:12)
```

To go over the issue I had to clean all path aliases from the _tsconfig.base.json_ at the root of the project which causes the project to fail to build afterwards.



I realize nx version is probably less interesting in the scope of this project but still...
```
 NX   Report complete - copy this into the issue template

Node           : 22.14.0
OS             : linux-x64
Native Target  : x86_64-linux
pnpm           : 9.1.0

nx                     : 20.4.6
@nx/js                 : 20.4.6
@nx/jest               : 20.4.6
@nx/eslint             : 20.4.6
@nx/workspace          : 20.4.6
@nx/angular            : 20.4.6
@nx/devkit             : 20.4.6
@nx/eslint-plugin      : 20.4.6
@nx/module-federation  : 20.4.6
@nx/node               : 20.4.6
@nx/rollup             : 20.4.6
@nx/web                : 20.4.6
@nx/webpack            : 20.4.6
typescript             : 5.4.5
---------------------------------------
Community plugins:
ngx-build-plus : 17.0.0
---------------------------------------
Local workspace plugins:
         [redacted]
```

**Reply by kdy1** [reply]
Can you file an issue?

**Reply by tonyganchev** [reply]
I actually realized this is already reported to the folks in nrwl: https://github.com/nrwl/nx/issues/28804#issuecomment-2687478476
I don't think ATM there's anything for the swc team to investigate. Nx have issues with too many of their codepaths ending in swc asserts rather than more reasonable error reporting.
